### PR TITLE
[docs] updated contribution.md with create local branch & directory structure

### LINKED ISSUE #2131: docs: update contributing guide
### Description
As a developer, I want to update the contributing guide with the following information
1. Update pull request process and add more details. Can take reference from [here](https://github.com/hyperledger/cactus/blob/main/CONTRIBUTING.md#create-local-branch)
2. Addition of commit lint workflow to streamline commit messages. Can take reference from [here](https://github.com/hyperledger/cactus/blob/main/.github/workflows/commitlint-pull-request.yml)

### Acceptance Criteria
1. Addition of commit lint workflow mandatory on all PR to dev and main branches
2. Updation of contributing guide.

### PR DESCRIPTION
**Pull request by maheshkasabe**
**Change log**

Update Contribution.md by adding create local branch & directory structure section

Fixes #2131

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -51,6 +51,79 @@ For source code integrity , Hyperledger Bevel GitHub pull requests are accepted
...
+
+NOTE: You can refer original tutorial ['GitHub Standard Fork & Pull Request Workflow'](https://gist.github.com/Chaser324/ce0505fbed06b947d962)
+
+### Directory structure
+
+Whenever you begin to use your codes on Hyperledger Bevel, you should follow the directory strecture on Hyperledger Bevel.
+The current directory structure is described as the following:
+
+
+> - automation/ : This folder contains the Input configuration file and Jenkinsfile for r3-corda, hyperledger fabric which helps in automation of corda/fabric network.
```
Comment (jagpreetsinghsasan) [inline CONTRIBUTING.md:105]: The automation folder structure description is totally wrong. Please check again what the Jenkins file contain.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -51,6 +51,79 @@ For source code integrity , Hyperledger Bevel GitHub pull requests are accepted
...
+>    - pip-requirements.txt
+>    - README.md     
+> - platforms/
+>     - hyperledger-besu/
+>     - hyperledger-fabric/ 
+>     - hyperledger-indy/ 
+>     - quorum/ 
+>     - r3-corda-ent/
+>     - r3-corda/ 
+>     - shared/ : The shared folder contains various sub folders which in turn each have certain types of files that will be used as prerequisites
```
Comment (jagpreetsinghsasan) [inline CONTRIBUTING.md:124]: The description of this is wrong.
Please change it to something like
`This folder contains the master playbook (site.yaml), along with some pre-requisite roles, playbooks and other shared roles which are commonly used by all the platforms`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -51,6 +51,79 @@ For source code integrity , Hyperledger Bevel GitHub pull requests are accepted
...
+   ```
+   git checkout <newfeature>
+   git rebase main
+   # Happy coding !
+   ```
+5. Commit changes to your branch.
+   ```
+   # Commit and push your changes to your fork
+   git add -A
+   git commit -s -m "<type>[optional scope]: <description>"
```
Comment (jagpreetsinghsasan) [inline CONTRIBUTING.md:88]: I would rather change it to something like

`git commit -s`
(This opens the default text editor window, where you can write the commit title and message, along with a `fixes #issue_number` line at the end)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -51,6 +51,79 @@ For source code integrity , Hyperledger Bevel GitHub pull requests are accepted
...
+   git fetch upstream
+   git checkout main
+   git rebase upstream/main
+   ```
+3. Create your branch.
+   ```
+   # Checkout the main branch - you want your new branch to come from main
+   git checkout main
+   # Create a new branch named `<newfeature>` (give simple informative name)
+   git branch <newfeature>
```
Comment (jagpreetsinghsasan) [inline CONTRIBUTING.md:76]: I would rather change it to
`git branch feat-issue_number` 
or
`git branch bug-issue_number`
or 
`git branch docs-issue_number`

In this way, we can create PR and commit titles accordingly in the future. For example:
If the issue is a feature for fabric, 
The branch can be `feat-000`
And then the PR title can be `[fabric] feat: added support for sample_feature`
And then the commit can be something like
```
feat(fabric): added support for sample_feature  
   Updated ....  
   Fixed .....  
Fixes #000
 ```

File: CONTRIBUTING.md (hunk 3 above)
Comment (jagpreetsinghsasan) [reply CONTRIBUTING.md:88]: See next review comment and you can added the commit title + body reference from there

**Comment by sownak** [comment]
@maheshkasabe Please fix the DCI lint error

**Review by sownak — APPROVED** [review]

**Pull request merged**